plugins/scaffolder-backend/src/scaffolder/stages/prepare/types.ts

```diff
@@ -25,7 +25,7 @@ export type PreparerBase = {
    */
   prepare(
     template: TemplateEntityV1alpha1,
-    opts: { logger: Logger },
+    opts?: { logger: Logger; workingDirectory?: string },
```

Comment:
@benjdlambert would this be ok? If I didn't make the `opts` optional I had to add a lot of `{}` as the second parameter in the tests.

Reply:
Do the tests not already pass in a logger? Unsure why you would have to pass `{}` if the opts are already required?

Reply:
The tests did not pass in a logger before. I guess it worked due to that the implementing classis didn't specify the `opts` in their signature.

```typescript
export class GithubPreparer implements PreparerBase {
  token?: string;

  constructor(params: { token?: string } = {}) {
    this.token = params.token;
  }

  async prepare(template: TemplateEntityV1alpha1): Promise<string> {
  }
};
```
Now I have added the `opts` parameter to their signature but the only property I have added to that is the optional `workindDirectory`.

```typescript
export class GithubPreparer implements PreparerBase {
  token?: string;

  constructor(params: { token?: string } = {}) {
    this.token = params.token;
  }

  async prepare(
    template: TemplateEntityV1alpha1,
    opts?: { workingDirectory?: string },
  ): Promise<string> {
  }
};
```

@benjdlambert any thoughts on what you want to do here? I would really like to get this into the next release 🙂

Reply:
I think this is all fine so far. Do you wanna convert it from draft when you're ready?

Reply:
I will do that now then. Should I write a changeset as well?

Reply:
Yep - be good to get into the habbit of it. Just make it a patch release for now though as we're`<1.0.0`

Reply:
Added changeset